src/plugins/kibana_utils/public/state_containers/types.ts

```diff
@@ -20,12 +20,13 @@
 import { Observable } from 'rxjs';
 import { Ensure, RecursiveReadonly } from '@kbn/utility-types';
 
+export type BaseState = Record<string, any>;
```

Comment:
Not sure if this is the right BaseState to use and if there would be any issues with this, but the practical reasoning relate to `syncState()` is:

```
import { encode } from 'rison-node'

export function test1<State extends object >(state: State) {
  encode(state) // doesn't work. ts complains that state is not assignable to RisonValue
}

export function test2<State extends Record<string, any>>(state: State) {
  encode(state) // works.
}

```

Reply:
I would not use `Record<string, any>` as that means state is basically any object, because key is any `string` and values of those keys are `any`. Maybe `object` can work here.